src/opendr/perception/object_tracking_3d/single_object_tracking/vpit/eval.py

```diff
@@ -0,0 +1,435 @@
+    dataset_tracking_path + "/training/velodyne/" + track_id,
+    dataset_tracking_path + "/training/label_02/" + track_id + ".txt",
+    dataset_tracking_path + "/training/calib/" + track_id + ".txt",
+)
+name = "pointpillars_car"
+config = all_configs[name]
+model_path = model_paths[name]
+
+
+tanet_name = "tanet_car"
+tanet_config = all_configs[tanet_name]
+tanet_model_path = model_paths[tanet_name]
+
+
+pq = 1
+lq = 20
+
+
+def tracking_boxes_to_camera(
+    label_original, calib,
+):
+
+    label = label_original.kitti()
+
+    if len(label["name"]) <= 0:
+        return label_original
+
+    r0_rect = calib["R0_rect"]
+    trv2c = calib["Tr_velo_to_cam"]
+
+    dims = label["dimensions"]
+    locs = label["location"]
+    rots = label["rotation_y"]
+
+    boxes_lidar = np.concatenate([locs, dims, rots.reshape(-1, 1)], axis=1)
+    boxes_camera = box_lidar_to_camera(boxes_lidar, r0_rect, trv2c)
+    locs_camera = boxes_camera[:, 0:3]
+    dims_camera = boxes_camera[:, 3:6]
+    rots_camera = boxes_camera[:, 6:7]
+
+    new_label = {
+        "name": label["name"],
+        "truncated": label["truncated"],
+        "occluded": label["occluded"],
+        "alpha": label["alpha"],
+        "bbox": label["bbox"],
+        "dimensions": dims_camera,
+        "location": locs_camera,
+        "rotation_y": rots_camera,
+        "score": label["score"],
+        "id": label["id"]
+        if "id" in label
+        else np.array(list(range(len(label["name"])))),
+        "frame": label["frame"]
+        if "frame" in label
+        else np.array([0] * len(label["name"])),
+    }
+
+    result = TrackingAnnotation3DList.from_kitti(
+        new_label, new_label["id"], new_label["frame"]
+    )
+
+    return result
+
+
+def label_to_AABB(label):
+
+    if len(label) == 0:
+        return label
+
+    label = label.kitti()
+
+    dims = label["dimensions"]
+    locs = label["location"]
+    rots = label["rotation_y"]
+
+    origin = [0.5, 0.5, 0]
+    gt_corners = center_to_corner_box3d(
+        locs, dims, rots.reshape(-1), origin=origin, axis=2,
+    )
+
+    mins = np.min(gt_corners, axis=1)
+    maxs = np.max(gt_corners, axis=1)
+    centers = (maxs + mins) / 2
+    sizes = maxs - mins
+    rotations = np.zeros((centers.shape[0],), dtype=np.float32)
+
+    new_label = {
+        "name": label["name"],
+        "truncated": label["truncated"],
+        "occluded": label["occluded"],
+        "alpha": label["alpha"],
+        "bbox": label["bbox"],
+        "dimensions": sizes,
+        "location": centers,
+        "rotation_y": rotations,
+        "score": label["score"],
+        "id": label["id"],
+        "frame": label["frame"],
+    }
+
+    result = TrackingAnnotation3DList.from_kitti(
+        new_label, new_label["id"], new_label["frame"]
+    )
+
+    return result
+
+
+def test_rotated_pp_siamese_eval(draw=True, iou_min=0.0, classes=["Car", "Van", "Truck"]):
+    print("Eval", name, "start", file=sys.stderr)
+    import pygifsicle
```

Comment:
This package seems to be missing from the dependencies. If it's installed anyway, ignore this comment.

Reply:
Updated dependencies